src/de.rs

```diff
@@ -579,9 +579,9 @@ impl<S: Read> de::Decoder for Decoder<S> {
             self.decode_map(visitor).await
         } else if self.numeric.contains(&self.buffer[0]) {
             self.decode_number(visitor).await
-        } else if self.buffer.len() >= FALSE.len() && self.buffer.starts_with(FALSE) {
-            self.decode_bool(visitor).await
-        } else if self.buffer.len() >= TRUE.len() && self.buffer.starts_with(TRUE) {
```

Comment:
would maybe want to make sure that a test hits this, but clippy essentially complained that since the two conditional blocks are the same,  the conditions should just be merged, so that's what i did here. the logic should be identical though.

Reply:
i could be wrong, but after trying to set some breakpoints and running the existing tests, it doesn't look like anything hits this condition.

Reply:
updated [this test here](https://github.com/haydnv/destream_json/pull/3/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R349), and confirmed it hits this code path.